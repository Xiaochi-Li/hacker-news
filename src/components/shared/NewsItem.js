import React from 'react';
import IconText from './IconText';
import TagGroup from './TagGroup';
import { List, Typography } from 'antd';
import { StarTwoTone, MessageTwoTone } from '@ant-design/icons';
import { htmlToText } from 'html-to-text';

const NewsItem = ({
  title,
  description,
  points,
  num_comments,
  url,
  story_text,
  _tags,
}) => {
  return (
    <List.Item
      key={title}
      actions={[
        <IconText text={points} icon={StarTwoTone} key="star-two-tone-icon" />,
        <IconText
          text={num_comments}
          icon={MessageTwoTone}
          key="message-two-tone-icon"
        />,
        <TagGroup tags={_tags} key="tag-group" />,
      ]}
    >
      <List.Item.Meta
        title={
          <a href={url} rel="noreferrer" target="_blank">
            {title}
          </a>
        }
        description={description}
      />
      <Typography.Paragraph
        ellipsis={{ rows: 3, expandable: true, symbol: 'more' }}
      >
        {htmlToText(story_text)}
      </Typography.Paragraph>
    </List.Item>
  );
};

export default NewsItem;
