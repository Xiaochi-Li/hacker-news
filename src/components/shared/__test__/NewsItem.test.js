import React from 'react';
import { render } from '@testing-library/react';
import NewsItem from '../NewsItem';

const testProps = {
  title: 'test title',
  description: 'test description',
  points: 100,
  num_comments: 200,
  url: 'https://www.test.com/',
  story_text: '<p>test story</p>',
  _tags: ['test tag 1', 'test tag 2'],
};

describe('<NewsItem/> spec', () => {
  it('renders NewsItem component', () => {
    const component = render(<NewsItem {...testProps} />);
    expect(component).toMatchSnapshot();
  });
});
