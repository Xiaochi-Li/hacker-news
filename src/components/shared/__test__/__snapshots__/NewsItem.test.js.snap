// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsItem/> spec renders NewsItem component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="ant-list-item"
      >
        <div
          class="ant-list-item-meta"
        >
          <div
            class="ant-list-item-meta-content"
          >
            <h4
              class="ant-list-item-meta-title"
            >
              <a
                href="https://www.test.com/"
                rel="noreferrer"
                target="_blank"
              >
                test title
              </a>
            </h4>
            <div
              class="ant-list-item-meta-description"
            >
              test description
            </div>
          </div>
        </div>
        <div
          class="ant-typography ant-typography-ellipsis"
        >
          test story
        </div>
        <ul
          class="ant-list-item-action"
        >
          <li>
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <span
                  aria-label="star"
                  class="anticon anticon-star"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="star"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512.5 190.4l-94.4 191.3-211.2 30.7 152.8 149-36.1 210.3 188.9-99.3 188.9 99.2-36.1-210.3 152.8-148.9-211.2-30.7z"
                      fill="#e6f7ff"
                    />
                    <path
                      d="M908.6 352.8l-253.9-36.9L541.2 85.8c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L370.3 315.9l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1L239 839.4a31.95 31.95 0 0046.4 33.7l227.1-119.4 227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM665.3 561.3l36.1 210.3-188.9-99.2-188.9 99.3 36.1-210.3-152.8-149 211.2-30.7 94.4-191.3 94.4 191.3 211.2 30.7-152.8 148.9z"
                      fill="#1890ff"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                100
              </div>
            </div>
            <em
              class="ant-list-item-action-split"
            />
          </li>
          <li>
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <span
                  aria-label="message"
                  class="anticon anticon-message"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="message"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M775.3 248.9a369.62 369.62 0 00-119-80A370.2 370.2 0 00512.1 140h-1.7c-99.7.4-193 39.4-262.8 109.9-69.9 70.5-108 164.1-107.6 263.8.3 60.3 15.3 120.2 43.5 173.1l4.5 8.4V836h140.8l8.4 4.5c52.9 28.2 112.8 43.2 173.1 43.5h1.7c99 0 192-38.2 262.1-107.6 70.4-69.8 109.5-163.1 110.1-262.7.2-50.6-9.5-99.6-28.9-145.8a370.15 370.15 0 00-80-119zM312 560a48.01 48.01 0 010-96 48.01 48.01 0 010 96zm200 0a48.01 48.01 0 010-96 48.01 48.01 0 010 96zm200 0a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
                      fill="#e6f7ff"
                    />
                    <path
                      d="M664 512a48 48 0 1096 0 48 48 0 10-96 0zm-400 0a48 48 0 1096 0 48 48 0 10-96 0z"
                      fill="#1890ff"
                    />
                    <path
                      d="M925.2 338.4c-22.6-53.7-55-101.9-96.3-143.3a444.35 444.35 0 00-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6.3-119.3 12.3-174.5 35.9a445.35 445.35 0 00-142 96.5c-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9A449.4 449.4 0 00112 714v152a46 46 0 0046 46h152.1A449.4 449.4 0 00510 960h2.1c59.9 0 118-11.6 172.7-34.3a444.48 444.48 0 00142.8-95.2c41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5.3-60.9-11.5-120-34.8-175.6zm-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-.6 99.6-39.7 192.9-110.1 262.7z"
                      fill="#1890ff"
                    />
                    <path
                      d="M464 512a48 48 0 1096 0 48 48 0 10-96 0z"
                      fill="#1890ff"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                200
              </div>
            </div>
            <em
              class="ant-list-item-action-split"
            />
          </li>
          <li>
            <div>
              <span
                class="ant-tag ant-tag-magenta"
              >
                test tag 1
              </span>
              <span
                class="ant-tag ant-tag-red"
              >
                test tag 2
              </span>
            </div>
          </li>
        </ul>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="ant-list-item"
    >
      <div
        class="ant-list-item-meta"
      >
        <div
          class="ant-list-item-meta-content"
        >
          <h4
            class="ant-list-item-meta-title"
          >
            <a
              href="https://www.test.com/"
              rel="noreferrer"
              target="_blank"
            >
              test title
            </a>
          </h4>
          <div
            class="ant-list-item-meta-description"
          >
            test description
          </div>
        </div>
      </div>
      <div
        class="ant-typography ant-typography-ellipsis"
      >
        test story
      </div>
      <ul
        class="ant-list-item-action"
      >
        <li>
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
              style="margin-right: 8px;"
            >
              <span
                aria-label="star"
                class="anticon anticon-star"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="star"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512.5 190.4l-94.4 191.3-211.2 30.7 152.8 149-36.1 210.3 188.9-99.3 188.9 99.2-36.1-210.3 152.8-148.9-211.2-30.7z"
                    fill="#e6f7ff"
                  />
                  <path
                    d="M908.6 352.8l-253.9-36.9L541.2 85.8c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L370.3 315.9l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1L239 839.4a31.95 31.95 0 0046.4 33.7l227.1-119.4 227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM665.3 561.3l36.1 210.3-188.9-99.2-188.9 99.3 36.1-210.3-152.8-149 211.2-30.7 94.4-191.3 94.4 191.3 211.2 30.7-152.8 148.9z"
                    fill="#1890ff"
                  />
                </svg>
              </span>
            </div>
            <div
              class="ant-space-item"
            >
              100
            </div>
          </div>
          <em
            class="ant-list-item-action-split"
          />
        </li>
        <li>
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
              style="margin-right: 8px;"
            >
              <span
                aria-label="message"
                class="anticon anticon-message"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="message"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M775.3 248.9a369.62 369.62 0 00-119-80A370.2 370.2 0 00512.1 140h-1.7c-99.7.4-193 39.4-262.8 109.9-69.9 70.5-108 164.1-107.6 263.8.3 60.3 15.3 120.2 43.5 173.1l4.5 8.4V836h140.8l8.4 4.5c52.9 28.2 112.8 43.2 173.1 43.5h1.7c99 0 192-38.2 262.1-107.6 70.4-69.8 109.5-163.1 110.1-262.7.2-50.6-9.5-99.6-28.9-145.8a370.15 370.15 0 00-80-119zM312 560a48.01 48.01 0 010-96 48.01 48.01 0 010 96zm200 0a48.01 48.01 0 010-96 48.01 48.01 0 010 96zm200 0a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
                    fill="#e6f7ff"
                  />
                  <path
                    d="M664 512a48 48 0 1096 0 48 48 0 10-96 0zm-400 0a48 48 0 1096 0 48 48 0 10-96 0z"
                    fill="#1890ff"
                  />
                  <path
                    d="M925.2 338.4c-22.6-53.7-55-101.9-96.3-143.3a444.35 444.35 0 00-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6.3-119.3 12.3-174.5 35.9a445.35 445.35 0 00-142 96.5c-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9A449.4 449.4 0 00112 714v152a46 46 0 0046 46h152.1A449.4 449.4 0 00510 960h2.1c59.9 0 118-11.6 172.7-34.3a444.48 444.48 0 00142.8-95.2c41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5.3-60.9-11.5-120-34.8-175.6zm-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-.6 99.6-39.7 192.9-110.1 262.7z"
                    fill="#1890ff"
                  />
                  <path
                    d="M464 512a48 48 0 1096 0 48 48 0 10-96 0z"
                    fill="#1890ff"
                  />
                </svg>
              </span>
            </div>
            <div
              class="ant-space-item"
            >
              200
            </div>
          </div>
          <em
            class="ant-list-item-action-split"
          />
        </li>
        <li>
          <div>
            <span
              class="ant-tag ant-tag-magenta"
            >
              test tag 1
            </span>
            <span
              class="ant-tag ant-tag-red"
            >
              test tag 2
            </span>
          </div>
        </li>
      </ul>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
